managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2789,29 +2789,22 @@ public void skipNonRecoverableLedger(final long ledgerId){
         if (ledgerInfo == null) {
             return;
         }
-        lock.writeLock().lock();
         log.warn("[{}] [{}] Since the ledger [{}] is lost and the autoSkipNonRecoverableData is true, this ledger will"
                 + " be auto acknowledge in subscription", ledger.getName(), name, ledgerId);
-        try {
-            for (int i = 0; i < ledgerInfo.getEntries(); i++) {
-                if (!individualDeletedMessages.contains(ledgerId, i)) {
```

Comment:
It looks like this condition was dropped is that intentional?

Reply:
https://github.com/apache/pulsar/blob/74f1cd51380b71c77d2f33ecb5830eab19ecb6f3/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java#L2280-L2291
Yes, because the condition will be checked again in the asyncDelete method, it can be removed here.

Reply:
@hrzzzz Thanks, makes sense.